pkg/redfish/redfish-foxconn.go

```diff
@@ -0,0 +1,323 @@
+		path := ncsiMember.Oid
+
+		url := s.epRF.FQDN + path
+		jsonData, err = s.epRF.GETRelative(path)
+		if err != nil || jsonData == nil {
+			s.LastStatus = HTTPsGetFailed
+			return
+		}
+		s.LastStatus = HTTPsGetOk
+
+		var nm InsydeOemNcsiMember
+		if err := json.Unmarshal(jsonData, &nm); err != nil {
+			errlog.Printf("Failed to decode %s: %s\n", url, err)
+			s.LastStatus = EPResponseFailedDecode
+		}
+
+		//////////////////////////////////////////////////////
+		// Parse each /redfish/v1/Systems/system/Oem/Insyde/Ncsi/# package member element
+		// Have only ever seen one package but we should iterate anyway in case that ever changes
+
+		for _, nmPkg := range nm.Package {
+
+			//////////////////////////////////////////////////////
+			// Parse /redfish/v1/Systems/system/Oem/Insyde/Ncsi/#/Package/#
+
+			path = nmPkg.Oid
+
+			url = s.epRF.FQDN + path
+			jsonData, err = s.epRF.GETRelative(path)
+			if err != nil || jsonData == nil {
+				s.LastStatus = HTTPsGetFailed
+				return
+			}
+			s.LastStatus = HTTPsGetOk
+
+			var p InsydeOemPackage
+			if err := json.Unmarshal(jsonData, &p); err != nil {
+				errlog.Printf("Failed to decode %s: %s\n", url, err)
+				s.LastStatus = EPResponseFailedDecode
+			}
+
+			//////////////////////////////////////////////////////
+			// Parse each /redfish/v1/Systems/system/Oem/Insyde/Ncsi/#/Package/#.PackageInfo[]
+
+			for j, pi := range p.PackageInfo {
+				// Only process if there is a MAC address
+				if pi.MACAddress != "" {
+					s.ENetInterfaces.Num++
+
+					ei := NewEpEthInterface(s.epRF, ncsiMember.Oid, s.RedfishSubtype, nmPkg, s.ENetInterfaces.Num)
+
+					ei.EtherIfaceRF.Oid = path
+					ei.EtherIfaceRF.MACAddress = pi.MACAddress
+					ei.EtherIfaceRF.Description = FOXCONN_ETH_INT_DESCRIPTION
+
+					// ID = "ncsi-" + ncsi number + "-" + package number + "-" + channel index
+					ei.EtherIfaceRF.Id = "ncsi" + nm.Id + "-p" + p.Id + "-c" + fmt.Sprint(j)
+
+					// According to Foxconn, this is the only unique identifier for the primary ethernet
+					if nm.VersionId.PCIDID == FOXCONN_PRIMARY_ETH_INT_PCIDID {
+						// We append a unique string to the end of the ID so that we can identify the
+						// primary ethernet interface later.
+						ei.EtherIfaceRF.Id += FOXCONN_PRIMARY_ETH_INT_SUFFIX
+					} else if strings.TrimSpace(nm.VersionId.FirmwareName) == FOXCONN_PRIMARY_ETH_INT_FW_NAME {
+						// Leave a breadcrumb if Foxconn ever changes the PCIDID for the primary ethernet device
+
+						errlog.Printf("Found suspect PCIDID=%s associated with FW \"%s\" for %s\n", nm.VersionId.PCIDID, nm.VersionId.FirmwareName, ei.EtherIfaceRF.Id)
+
+						// TODO: We could append FOXCONN_PRIMARY_ETH_SUFFIX to the ID as well if we (1) think
+						// Foxconn will ever change the PCIDID and (2) if we trust FOXCONN_PRIMARY_ETH_INT_FW_NAME
+						// will always be unique to the primary ethernet device.  When asked about a unique identifier
+						// they said to use the PCIDID.
```

Comment:
As noted in the comment above, if we were ok with the prior uniqueness issues surrounding FirmwareName then we could also append FOXCONN_PRIMARY_ETH_INT_SUFFIX to the Id here.  I don't feel the greatest about that but thought I'd mention it.